code/gui/mainwindow.cpp

```diff
@@ -58,19 +61,19 @@ void MainWindow::keyPressEvent(QKeyEvent *e) {
     m_controller->keyPressed(e->key());
     switch (e->key()) {
         case Qt::Key_Up:
-            m_controller->move(Controller::Dir::UP);
+            m_solenoid->queueInstruction(Controller::Dir::UP);

```

Comment:
This should still be calling `m_controller->move` which, in `Solenoid`, would call `queueInstruction` and then the execute move calls would happen in a separate slot called by the timer, otherwise the controller can't be changed to, say, the simulator